packages/back-end/src/types/Integration.ts

```diff
@@ -7,6 +7,22 @@ import { ExperimentInterface, ExperimentPhase } from "../../types/experiment";
 import { MetricInterface, MetricType } from "../../types/metric";
 import { SegmentInterface } from "../../types/segment";
 
+export class NoDefaultDatasetError extends Error {
+  constructor() {
+    super(
+      "To view the information schema for a BigQuery dataset, you must define a default dataset. Please add a default dataset by editing the datasource's connection settings."
```

Comment:
Don't hard-code a message here.  Pass it in as an argument.  Possibly make the error class name more generic to work for other data sources.